code/datums/components/heirloom.dm

```diff
@@ -9,16 +12,30 @@
 	owner = new_owner
 	family_name = new_family_name
 
-	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/examine)
+	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/on_examine)
 
-/datum/component/heirloom/proc/examine(datum/source, mob/user, list/examine_list)
+/datum/component/heirloom/Destroy(force, silent)
+	owner = null
+	return ..()
+
+/**
+ * Signal proc for [COMSIG_PARENT_EXAMINE].
+ *
+ * Shows who owns the heirloom on examine.
+ */
+/datum/component/heirloom/proc/on_examine(datum/source, mob/user, list/examine_list)
 	SIGNAL_HANDLER
 
-	var/datum/antagonist/obsessed/creeper = user.mind.has_antag_datum(/datum/antagonist/obsessed)
+	if(!user.mind || isobserver(user))
+		examine_list += span_notice("It is the [family_name] family heirloom, belonging to [owner].")
+		return
 
 	if(user.mind == owner)
 		examine_list += span_notice("It is your precious [family_name] family heirloom. Keep it safe!")
-	else if(creeper && creeper.trauma.obsession == owner)
+		return
+
+	var/datum/antagonist/obsessed/our_creeper = user.mind.has_antag_datum(/datum/antagonist/obsessed)
+	if(our_creeper?.trauma.obsession == owner)
```

Comment:
My gut says that it'd be cleaner to store the user's mind in a var, use ?. for it, and move the observer check to the creeper line, rather then duping the code

Reply:
Sure thing, lemmie know if that's what you had in mind

Reply:
👍